aptos-move/block-executor/src/executor.rs

```diff
@@ -199,6 +210,40 @@ where
         }
     }
 
+    fn commit_hook(
+        &self,
+        txn_idx: TxnIndex,
+        versioned_cache: &MVHashMap<T::Key, T::Value, ExecutableTestType>,
+        last_input_output: &TxnLastInputOutput<T::Key, E::Output, E::Error>,
+        base_view: &S,
+    ) {
+        let (num_deltas, delta_keys) = last_input_output.delta_keys(txn_idx);
+        let mut delta_writes = Vec::with_capacity(num_deltas);
+        for k in delta_keys {
+            let committed_delta = versioned_cache
+                .materialize_delta(&k, txn_idx)
```

Comment:
this is a bit ugly in the sense that the first time it's guaranteed to fail without storage value and needs to read and re-materialize again, maybe we should just check if aggregator_base is set and set it before calling materialize?

Reply:
Issue is overall it will be way less efficient because the base needs to be set once and materializing doesn't necessarily need to check it, most of the time it will just succeed based on some earlier materialized write. It's just when it fails without resolving we know base must have been missing and we can set it.. In other words, if we always checked, we would do ~2 times more locks overall than now - may fail in the beginning, but later everything works without even getting to the aggregator base.

Reply:
done, according to George's suggestion, no need to materialize twice.